docs/python/how-to-guides/sysadmin/setuptools-deployment.md

```diff
@@ -0,0 +1,407 @@
+my-dh-report = "my_dh_tool.reporter:generate_report"
+```
+
+### Advanced argument parsing
+
+For more sophisticated CLIs, consider using [typer](https://typer.tiangolo.com/) or [click](https://click.palletsprojects.com/):
+
+```python syntax
+import typer
+from typing_extensions import Annotated
+
+
+def app(
+    input_file: Annotated[str, typer.Argument(help="Input file to process")],
+    output: Annotated[str, typer.Option(help="Output file")] = None,
+    verbose: Annotated[bool, typer.Option(help="Verbose output")] = False,
+):
+    """Process data with Deephaven."""
+    if verbose:
+        typer.echo(f"Processing {input_file}...")
+
+    # Process data using Deephaven
+    result = process_data(input_file)
+
+    if output:
+        typer.echo(f"Results saved to {output}")
+
+
+if __name__ == "__main__":
+    typer.run(app)
+```
+
+Add `typer` to your dependencies:
+
+```toml
+dependencies = [
+  "deephaven-server>=0.35.0",
+  "typer>=0.9.0",
+]
+```
+
+### Configuration files
+
+For more complex tools, consider using configuration files:
+
+```python syntax
+import configparser
+import os
+
+
+def load_config():
+    config = configparser.ConfigParser()
+    config_path = os.path.expanduser("~/.my_dh_tool/config.ini")
+
+    if os.path.exists(config_path):
+        config.read(config_path)
+
+    return config
+```
+
+## Distribution
+
+### Distribute via PyPI
+
+Once your package is ready, you can publish it to the Python Package Index (PyPI):
+
+```shell
+python -m pip install twine
+python -m twine upload dist/*
+```
+
+Users can then install your tool with:
+
+```shell
+pip install my-dh-tool
+```
+
+### Distribute as wheel files
+
+Alternatively, distribute the `.whl` file directly to users who can install it with:
+
+```shell
+pip install my_dh_tool-0.1.0-py3-none-any.whl
+```
+
+## Best practices
+
+- **Version pinning**: Specify minimum Deephaven versions in the `dependencies` field to ensure compatibility.
+- **Error handling**: Add robust error handling for file operations and data processing.
+- **Use src-layout**: Follow the recommended src-layout structure for better package organization.
+- **Logging**: Use Python's `logging` module instead of `print()` for production tools.
+- **Testing**: Include unit tests for your command-line scripts.
+- **Documentation**: Provide clear usage instructions in your README and command-line help text.
+- **Virtual environments**: Develop and test in isolated virtual environments to avoid dependency conflicts.
+
+## Troubleshooting
+
+### Command not found after installation
+
+If your command isn't found after installation:
+
+- Ensure the installation completed successfully without errors.
+- Check that the installation directory is in your system's PATH.
+- Try reinstalling with `pip install --force-reinstall`.
+
+### Import errors
+
+If you encounter import errors when running your command:
+
+- Verify all dependencies are listed in the `dependencies` field of pyproject.toml.
```

Comment:
pyproject.toml is neither backticked or linked.

Can we link to a section of the python docs for `dependencies`?